sesman/chansrv/smartcard_pcsc.c

```diff
@@ -1686,18 +1689,30 @@ scard_process_cmd_wait_reader_state_change(struct trans *con,
                                            struct stream *in_s)
 {
     int rv;
-    //struct stream *out_s;
-    int timeOut;
+    int timeOut __attribute__((unused));
```

Comment:
On RHEL, without this dircective, I have the following error :
```
smartcard_pcsc.c: In function âscard_process_cmd_wait_reader_state_changeâ:
smartcard_pcsc.c:1688:9: error: variable âtimeOutâ set but not used [-Werror=unused-but-set-variable]
     int timeOut;
```

Reply:
Sorry - you are of course quite correct, and this comes about because of the `LOG_DEVEL()` macro.

We've got a temporary macro you can use to do this, rather than the attribute marking. Can you try the following:-

```c
int timeOut;
. . 
UNUSED_VAR(timeOut);
```
If you grep through for `UNUSED_VAR` you'll find a couple of uses of it.

Even better would be to  move the declarations closer to the point-of-use (i.e.) :-

```c
    int rv;
    <some declarations deleted here>
    struct pcsc_uds_client *uds_client;

    LOG_DEVEL(LOG_LEVEL_DEBUG, "scard_process_cmd_wait_reader_state_change:");
    uds_client = (struct pcsc_uds_client *) (con->callback_data);
    uds_client->waiting = 1;
    if ((uds_client->vMajor == 4) && (uds_client->vMinor > 2)) {
        //In these versions, return reader states immediately
       struct stream *out_s = con->out_s;
        int reader_state_bytes = sizeof(uds_client->readerStates);
        init_stream(out_s, reader_state_bytes);
        out_uint8a(out_s, uds_client->readerStates, reader_state_bytes);
        s_mark_end(out_s);
        rv = trans_write_copy(con);
    } else {
        int timeOut;
#ifndef USE_DEVEL_LOGGING
        /* TODO: remove UNUSED_VAR once the `timeOut` variable is used for more
        than logging in debug mode */
        UNUSED_VAR(timeOut);
#endif
        in_uint32_le(in_s, timeOut);
        LOG_DEVEL(LOG_LEVEL_DEBUG, "scard_process_cmd_wait_reader_state_change: timeOut %d",
           timeOut);
        in_uint32_le(in_s, rv);
        LOG_DEVEL(LOG_LEVEL_DEBUG, "scard_process_cmd_wait_reader_state_change: rv %d", rv);
    }
```

It's not pretty, but it's the same as we've used elsewhere.

Reply:
Thanks, modified with UNUSED_VAR macro.